### ISSUE TITLE
Mutator methods protected and final GaussianMutator prevent extending

### ISSUE THREAD

**Issue by vilppuvuorinen** [jenetics/jenetics#426]
Many Mutator methods are marked protected preventing extension by composition. It is not possible to forward method calls to another mutator. The less intriguing option would be to extend GaussianMutator, but it's marked as final for no reason at all.

Of course I can work around this with the good old Ctrl+C & Ctrl+V, but unnecessarily restricted access modifiers are simply odd.

**Comment by jenetics** [comment]
Which kind of extension points do you have in mind? It might be useful to implemented this with some static factory methods:
```java
Mutator<DoubleGene, Double> mutator = Mutator.of(...);
```

**Comment by vilppuvuorinen** [comment]
The use case I was going after was along the lines of '`GaussianMutator` that does specific normalization for each `Chromosome`'.

I first tried to go for a generic composed mutator that would be constructed using `new NormalizingMutator(new GaussianMutator())`. I would have then overridden the `protected` mutation method for `Chromosome` with my own implementation calling the mutator passed to constructor and doing the specific normalization before returning. This did not work out due to the methods being `protected`.

Next I tried to extend GaussianMutator which turned out to be final.

I do understand that `Mutator` instances are not guaranteed to call all mutation methods of `Mutator`. Implementations relying on this would not be guaranteed to work with all possible mutators and it could break if implementation of a mutator changes. This is probably at least a part of the reason these methods are protected.

Generic `Chromosome` post processor for `Mutator` would be possible to implement if gene level mutators would be defined as a separate class `GeneMutator`. This does not seem practical in the short term. Non-final `GaussianMutator` would go a long way tho.
